eos/utils/parameters.cc

```diff
@@ -754,6 +763,27 @@ namespace eos
         // ... before returning it
         return _imp->parameters.back();
     }
+
+    void
+    Parameters::redirect(const QualifiedName & name, const Parameter::Id & id)
+    {
+        ParameterDefaults::instance()->redirect(name, id);
+    }
+
+    void
+    Parameters::redirect_and_apply(const QualifiedName & name, const Parameter::Id & id)
+    {
+        // redirect the parameter to the new id ...
+        ParameterDefaults::instance()->redirect(name, id);
+
+        // ... and apply the change to this parameter set ...
+        auto i = _imp->parameters_map.find(name);
+        if (_imp->parameters_map.end() == i)
+            throw UnknownParameterError(name);
+
+        i->second = id;
```

Comment:
If I understand correctly, you need both because you are afraid that new parameter sets could be created from the default in the scope of the current analysis?

Reply:
So question: is there any case where we need `redirect` without `apply`?